python/pyspark/sql/functions.py

```diff
@@ -2301,10 +2301,12 @@ def pandas_udf(f=None, returnType=None, functionType=None):
        The returned scalar can be either a python primitive type, e.g., `int` or `float`
        or a numpy data type, e.g., `numpy.int64` or `numpy.float64`.
 
-       :class:`ArrayType`, :class:`MapType` and :class:`StructType` are currently not supported as
-       output types.
+       :class:`MapType` and :class:`StructType` are currently not supported as output types.
```

Comment:
@icexelloss, actually should we keep this note? I think this is matched with https://spark.apache.org/docs/latest/sql-programming-guide.html#supported-sql-types which we documented there and SQLConf.

Probably, just leaving a link could be fine. Removing out is okay to me too.

Reply:
I am leaning towards keeping this in the API doc and maybe make sql-programming-guide link to this.

I think most user would look for API docs first rather than sql-programming-guide, so it's probably a bit more convenient to have it here?

Reply:
Track in: https://issues.apache.org/jira/browse/SPARK-23633